pkg/controller/operators/olm/requirements.go

```diff
@@ -72,118 +71,9 @@ func (a *Operator) minKubeVersionStatus(name string, minKubeVersion string) (met
-		crd, err := a.lister.APIExtensionsV1beta1().CustomResourceDefinitionLister().Get(r.Name)
-		if err != nil {
-			status.Status = v1alpha1.RequirementStatusReasonNotPresent
-			status.Message = "CRD is not present"
-			a.logger.Debugf("Setting 'met' to false, %v with status %v, with err: %v", r.Name, status, err)
-			met = false
-			statuses = append(statuses, status)
-			continue
-		}
-
-		if crd.Spec.Version != r.Version {
-			served := false
-			for _, version := range crd.Spec.Versions {
-				if version.Name == r.Version {
-					if version.Served {
-						served = true
-					}
-					break
-				}
-			}
-
-			if !served {
-				status.Status = v1alpha1.RequirementStatusReasonNotPresent
-				status.Message = "CRD version not served"
-				a.logger.Debugf("Setting 'met' to false, %v with status %v, CRD version %v not found", r.Name, status, r.Version)
-				met = false
-				statuses = append(statuses, status)
-				continue
-			}
-		}
-
-		// Check if CRD has successfully registered with k8s API
-		established := false
-		namesAccepted := false
-		for _, cdt := range crd.Status.Conditions {
-			switch cdt.Type {
-			case v1beta1.Established:
-				if cdt.Status == v1beta1.ConditionTrue {
-					established = true
-				}
-			case v1beta1.NamesAccepted:
-				if cdt.Status == v1beta1.ConditionTrue {
-					namesAccepted = true
-				}
-			}
-		}
-
-		if established && namesAccepted {
-			status.Status = v1alpha1.RequirementStatusReasonPresent
-			status.Message = "CRD is present and Established condition is true"
-			status.UUID = string(crd.GetUID())
-			statuses = append(statuses, status)
-		} else {
-			status.Status = v1alpha1.RequirementStatusReasonNotAvailable
-			status.Message = "CRD is present but the Established condition is False (not available)"
-			met = false
-			a.logger.Debugf("Setting 'met' to false, %v with status %v, established=%v, namesAccepted=%v", r.Name, status, established, namesAccepted)
-			statuses = append(statuses, status)
-		}
-	}
-
-	// Check for required API services
-	for _, r := range requiredAPIServiceDescs {
-		name := fmt.Sprintf("%s.%s", r.Version, r.Group)
-		status := v1alpha1.RequirementStatus{
-			Group:   "apiregistration.k8s.io",
-			Version: "v1",
-			Kind:    "APIService",
-			Name:    name,
-		}
-
-		// Check if GVK exists
-		if err := a.isGVKRegistered(r.Group, r.Version, r.Kind); err != nil {
-			status.Status = "NotPresent"
-			met = false
-			statuses = append(statuses, status)
-			continue
-		}
-
-		// Check if APIService is registered
-		apiService, err := a.lister.APIRegistrationV1().APIServiceLister().Get(name)
-		if err != nil {
-			status.Status = "NotPresent"
-			met = false
-			statuses = append(statuses, status)
-			continue
-		}
-
-		// Check if API is available
-		if !a.isAPIServiceAvailable(apiService) {
-			status.Status = "NotPresent"
-			met = false
-		} else {
-			status.Status = "Present"
-			status.UUID = string(apiService.GetUID())
-		}
-		statuses = append(statuses, status)
-	}
-
```

Comment:
Before - a check for the APIService was added in addition to a GVK check - could you explain why this is not necessary anymore?

Reply:
in addition to checking the GVK of an APIService, this used to check is a APIService resource exist and if it is available. If the GVK of that APIService exist, shouldn't that imply that the APIService is available? If it is available, why do we care if the APISewrvice resource is there or not, since, for required APIService, we just care if the APIService is available to serve us. Right?

Reply:
> In addition to checking the GVK of an APIService, this used to check is a APIService resource exist and if it is available. I. If the GVK of that APIService exist, shouldn't that imply that the APIService is available?

I do not believe that this handles upgrades. Say that the deployment that implements the APIService is taken down and redeployed - although the GVK exists, the APIService would be unavailable during the duration of the upgrade. If the operator expects the APIService to be available, this would introduce an issue.